server/models/User.js

```diff
@@ -11,35 +11,36 @@ module.exports = (sequelize, DataTypes) => {
         validate: {
           len: {
             args: [3, 20],
-            msg: '"username" length must be between 3 and 20 characters long'
-          }
-        }
+            msg: '"username" length must be between 3 and 20 characters long',
+          },
+        },
       },
       email: {
         type: DataTypes.STRING,
         unique: true,
         allowNull: false,
-        validate: { isEmail: { msg: '"email" must be a valid email' } }
+        validate: { isEmail: { msg: '"email" must be a valid email' } },
       },
       hash: { type: DataTypes.STRING, allowNull: false },
       bio: DataTypes.TEXT,
       verified: {
         type: DataTypes.BOOLEAN,
         allowNull: false,
-        defaultValue: false
+        defaultValue: false,
       },
       image: {
         type: DataTypes.STRING,
-        validate: { isUrl: { msg: '"image" must be a valid URL' } }
-      }
+        validate: { isUrl: { msg: '"image" must be a valid URL' } },
+      },
     },
     {}
   );
   User.beforeValidate(user => {
-    user.hash = hashPassword(user.hash);
+    if (user.hash) user.hash = hashPassword(user.hash);
```

Comment:
I think the aim is to avoid rehashing an already hashed password. I would advise to confirm that `user.hash` only exists here when the field is created/updated.